File: Swift/Easy/Best-Time-to-Buy-and-Sell-Stock-II.swift
class Solution {
    func maxProfit(_ prices: [Int]) -> Int {
        
        guard !(prices.count == 1) else {
            return 0
        }
        
        var profits = 0
        
        for i in 1..<prices.count {
            profits += max(prices[i]-prices[i-1],0)
        }
        
        return profits
    }
}

File: CPP/Easy/Best-Time-to-Buy-and-Sell-Stock-II.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        if (prices.size() == 1) {
            return 0;
        }
        
        int profits = 0;
        
        for (int i = 0; i < prices.size() - 1; i++) {
            if (prices[i+1] > prices[i]) {
                profits += prices[i+1] - prices[i];
            }
        }
        return profits;
    }
};